aeron-samples/src/main/java/uk/co/real_logic/aeron/samples/MultipleSubscribersWithFragmentAssembly.java

```diff
@@ -0,0 +1,201 @@
+
+/**
+ * A subscriber application with two subscriptions which can receive fragmented messages
+ * Creates two subscriptions on a given CHANNEL which can receive two independent streams of messages.
+ * The default STREAM_ID and CHANNEL is
+ * is configured at {@link SampleConfiguration}. The default
+ * channel and stream can be changed by setting java system properties at the command line.
+ * i.e. (-Daeron.sample.channel=udp://localhost:5555 -Daeron.sample.streamId=20)
+ */
+public class MultipleSubscribersWithFragmentAssembly
+{
+    // A unique stream id within a given channel for subscriber one
+    private static final int STREAM_ID_1 = SampleConfiguration.STREAM_ID;
+    // A unique stream id within a given channel for subscriber two
+    private static final int STREAM_ID_2 = SampleConfiguration.STREAM_ID + 1;
+    // Channel ID for the application
+    private static final String CHANNEL = SampleConfiguration.CHANNEL;
+
+    // Maximum number of fragments to be read in a single poll operation
+    private static final int FRAGMENT_COUNT_LIMIT = SampleConfiguration.FRAGMENT_COUNT_LIMIT;
+
+    public static void main(final String[] args) throws Exception
+    {
+        System.out.println("Subscribing to " + CHANNEL + " on stream Id " + STREAM_ID_1 + " and stream Id " + STREAM_ID_2);
+
+        // Create a context for a client and specify callback methods when
+        // a new connection starts (eventNewConnection)
+        // a connection goes inactive (eventInactiveConnection)
+        final Aeron.Context ctx = new Aeron.Context()
+            .newConnectionHandler(MultipleSubscribersWithFragmentAssembly::eventNewConnection)
+            .inactiveConnectionHandler(MultipleSubscribersWithFragmentAssembly::eventInactiveConnection);
+
+        // dataHandler method is called for every new datagram received
+        // When a message is completely reassembled, the delegate method 'reassembledStringMessage' is called
+        final FragmentAssemblyAdapter dataHandler1 = new FragmentAssemblyAdapter(reassembledStringMessage1(STREAM_ID_1));
+
+        // Another Data handler for a different stream
+        final FragmentAssemblyAdapter dataHandler2 = new FragmentAssemblyAdapter(reassembledStringMessage2(STREAM_ID_2));
+
+        final AtomicBoolean running = new AtomicBoolean(true);
+
+        // Register a SIGINT handler
+        SigInt.register(() -> running.set(false));
+
+        // Create an Aeron instance with client provided context configuration and connect to media driver
+        try (final Aeron aeron = Aeron.connect(ctx);
+                // Add a subscription to Aeron for a given channel and steam. Also,
+                // supply a dataHandler to be called when data arrives
+                final Subscription subscription1 = aeron.addSubscription(CHANNEL, STREAM_ID_1, dataHandler1);
+                final Subscription subscription2 = aeron.addSubscription(CHANNEL, STREAM_ID_2, dataHandler2))
+                {
```

Comment:
The spacing and formatting should be consistent with the rest of the Aeron code base.